libs/grok/src/test/java/org/elasticsearch/grok/GrokTests.java

```diff
@@ -439,4 +439,13 @@ public void testExponentialExpressions() {
         run.set(false);
         assertThat(e.getMessage(), equalTo("grok pattern matching was interrupted after [200] ms"));
     }
+
+    public void testUnicodeFieldnames() {
```

Comment:
@talevy as far as I know the `[:alphanum:]` range isn't contiguous (it def. isn't at the boundary between letters and numbers I think).
So:

> add tests for fieldnames with randomAlphas & randomInts to ensure that the existing A-z0-9 will continue to work

seems the way to go imo (obviously could choose a different range as well but I don't think it would be logically different from just testing the old range + 2 concrete examples of things that broke before).

Are you sure we want to test the negative case here too? So basically just use some char that we def don't support in the filename like say `(field` and match for `null`?

Reply:
just as a sanity check that declares we do not support arbitrary unicode. I don't think we have that around

Reply:
@talevy sure, will add that (though come to think of it, the behaviour there kind of sucks in that we're just quietly returning `null` for fieldnames that can't even work to begin with).

I wonder if we should maybe add some validation for the `grokPattern` to the `Grok` constructor (not necessarily in this PR ... more thinking in general)?

Reply:
> the behaviour there kind of sucks in that we're just quietly returning null for fieldnames that can't even work to begin 

This definitely sounds like something that should be fixed in a followup.